### DISCUSSION TITLE
Is there a cross platform way to achieve mouse penetration?

### DISCUSSION THREAD

**Discussion by metaone01** [AvaloniaUI/Avalonia#17836] [Q&A] [answered]
I have been searching for ways to achieve **mouse penetration**, but all the blogs use winform solutions(like *user32.dll*) that are only applicable on the Windows platform.
Is there any feature to achieve this easily? Or do we have to use hooks?

**Comment by thevortexcloud** [comment, marked as answer]
This question has been asked several times. There is no built in way in Avalonia to do this. It _can_ be done though if you are willing to write the native interop code yourself. Avalonia generally exposes things like the native window handle which should be all you need to modify the window behaviour with native APIs. But not all platforms necessarily allow for clicking through windows. 

For Linux you can use the XShape API. But there are almost no C# examples for it and the examples that do exist are in C++/C.

https://en.wikipedia.org/wiki/Shape_extension

For Windows you just need to set some window flags from memory. But I don't remember which ones.

https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowlonga

For Mac I have no idea where to even start with that.

**Reply by metaone01** [reply]
These are very helpful to me. Thank you!
